### DISCUSSION TITLE
TIling CMYK PDFs that have transparent background

### DISCUSSION THREAD

**Discussion by viktor-meglenovski** [ImageMagick/ImageMagick#7367] [Help]
### ImageMagick version

7.1.1-12

### Operating system, version and so on

Windows 10

### Description

Hello everyone,

I am struggling for a long time trying to tile multiple PDFs that are in CMYK color space and have transparent backgrounds on a big canvas that is also CMYK. The main issue that I have is that even though the PDFs have a transparent background when I open them in Photoshop, ImageMagick loads the background as white. A possible workaround for me was to replace all pure white pixels with transparent, but this does not work every time since some of the PDFs may have pure white pixels themselves (resulting in them becoming transparent as well).

Lately I was thinking of workaround with copying the alpha channel of the RGB version of the PDF and composing it over the CMYK version of the same PDF.

First I tried the following command, and I can say that the alpha channel is extracting as it should from the RGB image. Moreover the resulting PDF is also in CMYK and the transparency is preserved:
magick convert a.pdf ( -colorspace RGB a.pdf -alpha extract ) -compose CopyOpacity -composite b.pdf

Here is my command:
magick -size 8400x6300 xc:none -units PixelsPerInch -density 300 -colorspace CMYK ( ( a.pdf ( -colorspace RGB a.pdf -alpha extract ) -compose CopyOpacity -composite ) -density 300 -resize 2480x2480 -rotate +270 ) -geometry +6175.4+4506.2 -composite b.pdf

However, this results in preserving the transparency, but the parts of the image that should not be transparent, are filled with plain #231f20 color. I am not sure why the actual colors are not shown.

Note: I have previously tried using -background None, -alpha set and similar methods but as I understood GhostScript cannot really process transparency in CMYK images. That is why I started thinking about extracting the alpha channel from the RGB.

Any help is appreciated.

Thank you all!

**Comment by fmw42** [comment]
On IM 7 use magick, not magick convert.

Imagemagick uses Ghostscript to read PDF files. Unfortunately, it will not read CMYK with alpha.  So you have to convert the CMYK to sRGB in order to keep the alpha.

`magick -density XXX -colorspace sRGB image.pdf ...... result.pdf`

I am not sure if Imagemagick can write CMYK with alpha. You can try since it does not use Ghostscript to write PDF.  However, when you read PDF, Ghostscript will rasterize any vector components and when you write back out even to PDF you will lose those vectors to pixel equivalent.

`magick -density XXX -colorspace sRGB image.pdf ...... -colorspace CMYK result.pdf`

or better use profiles

`magick -density XXX -colorspace sRGB image.pdf ...... -profile sRGB.icc -profile USWebCoatedSWOP.icc result.pdf`

You will have to use your own paths to the profiles, etc.

**Reply by viktor-meglenovski** [reply]
Thank you for your reply. Actually I do not have problems with writing the final PDF, that PDF has transparent background and is CMYK. The problem is when I place the individual PDFs on the canvas, I am not sure why CopyOpacity copies the exact alpha channel as it should (the background is removed) but I am left with the shape of the object filled with #231f20.

Note: magick convert a.pdf ( -colorspace RGB a.pdf -alpha extract ) -compose CopyOpacity -composite b.pdf
this works well, but when I try to place it on the canvas the problem I wrote about above appears.

**Reply by fmw42** [reply]
How are you placing it on a canvas?  What canvas?

**Reply by viktor-meglenovski** [reply]
In the command I wrote above, first i use
`-size 8400x6300 xc:none -units PixelsPerInch -density 300 -colorspace CMYK`
and then for each individual PDF that I have I use 
`-geometry +6175.4+4506.2 -composite`
to place it on the desired coordinates.

**Reply by fmw42** [reply]
Please show complete command lines, not pieces of them.  I need to know exactly how you process the PDF and how you put it on your canvas.

**Reply by viktor-meglenovski** [reply]
This is the final command:
`magick -size 8400x6300 xc:none -units PixelsPerInch -density 300 -colorspace CMYK ( ( a.pdf ( -colorspace RGB a.pdf -alpha extract ) -compose CopyOpacity -composite ) -density 300 -resize 2480x2480 -rotate +270 ) -geometry +6175.4+4506.2 -composite b.pdf`

in this example I am placing only one PDF on the final canvas (a.pdf), ideally I would have multiple PDFs that I place on different locations on the canvas, I have that part tested and that works well, for sake of simplicity in this command I have supplied only 1 PDF.

Thank you
